src/config.rs

```diff
@@ -172,65 +173,113 @@ impl IdentFormat {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize), serde(default))]
 pub struct IdentFormats(HashMap<String, IdentFormat>);
 
-impl Default for IdentFormats {
-    fn default() -> Self {
-        let mut map = HashMap::new();
-
-        map.insert("field_accessor".into(), IdentFormat::default().snake_case());
-        map.insert(
-            "field_reader".into(),
-            IdentFormat::default().constant_case().suffix("_R"),
-        );
-        map.insert(
-            "field_writer".into(),
-            IdentFormat::default().constant_case().suffix("_W"),
-        );
-        map.insert(
-            "enum_name".into(),
-            IdentFormat::default().constant_case().suffix("_A"),
-        );
-        map.insert(
-            "enum_write_name".into(),
-            IdentFormat::default().constant_case().suffix("_AW"),
-        );
-        map.insert("enum_value".into(), IdentFormat::default().constant_case());
-        map.insert(
-            "enum_value_accessor".into(),
-            IdentFormat::default().snake_case(),
-        );
-        map.insert("interrupt".into(), IdentFormat::default().constant_case());
-        map.insert("cluster".into(), IdentFormat::default().constant_case());
-        map.insert(
-            "cluster_accessor".into(),
-            IdentFormat::default().snake_case(),
-        );
-        map.insert("cluster_mod".into(), IdentFormat::default().snake_case());
-        map.insert("register".into(), IdentFormat::default().constant_case());
-        map.insert(
-            "register_spec".into(),
-            IdentFormat::default().pascal_case().suffix("_SPEC"),
-        );
-        map.insert(
-            "register_accessor".into(),
-            IdentFormat::default().snake_case(),
-        );
-        map.insert("register_mod".into(), IdentFormat::default().snake_case());
-        map.insert("peripheral".into(), IdentFormat::default().constant_case());
-        map.insert(
-            "peripheral_singleton".into(),
-            IdentFormat::default().constant_case(),
-        );
-        map.insert("peripheral_mod".into(), IdentFormat::default().snake_case());
-        map.insert(
-            "peripheral_feature".into(),
-            IdentFormat::default().snake_case(),
-        );
-
-        Self(map)
+impl IdentFormats {
+    pub fn new_theme() -> Self {
+        Self(HashMap::from([
+            ("field_accessor".into(), IdentFormat::default().snake_case()),
+            (
+                "field_reader".into(),
+                IdentFormat::default().pascal_case().suffix("R"),
+            ),
+            (
+                "field_writer".into(),
+                IdentFormat::default().pascal_case().suffix("W"),
+            ),
+            ("enum_name".into(), IdentFormat::default().pascal_case()),
```

Comment:
I'm not a fan of the verbosity of the code here. Also quite a bit of code duplication. Is there a way to minimize it?

Reply:
Is now better?

Reply:
yep, I like it much better now